### PR TITLE
test zenodo with elasticsearch v5.x #974

### PR DESCRIPTION
**Pull request by rusrushal13**
plugged in v5.2.2 es and green ticks on travis CI

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -53,22 +53,28 @@ services:
...
+  - REQUIREMENTS=prod E2E="no" SQLALCHEMY_DATABASE_URI="sqlite:///zenodo.db" ES_URL="https://download.elastic.co/elasticsearch/release/org/elasticsearch/distribution/tar/elasticsearch/2.2.0/elasticsearch-2.2.0.tar.gz"
+  - REQUIREMENTS=prod E2E="yes" SQLALCHEMY_DATABASE_URI="postgresql+psycopg2://postgres@localhost:5432/zenodo" ES_URL="https://download.elastic.co/elasticsearch/release/org/elasticsearch/distribution/tar/elasticsearch/2.2.0/elasticsearch-2.2.0.tar.gz"
+
+jdk:
+  - "oraclejdk8"  
 
 python:
   - "2.7"
   - "3.5"
+  - "pypy"
```
Comment (lnielsen) [inline .travis.yml]: We not going to run on pypy so you can remove this as it will add extra minutes to the build.

**Comment by lnielsen** [comment]
If you look in the Travis build, then ES5 installation is failing with:

```console
$ /tmp/elasticsearch/bin/plugin install -b mapper-attachments
/home/travis/build.sh: line 62: /tmp/elasticsearch/bin/plugin: No such file or directory
The command "/tmp/elasticsearch/bin/plugin install -b mapper-attachments" failed and exited with 127 during 
```

There's two issues:

- The CLI ``plugin`` in 2.x was renamed to ``elasticsearch-plugin`` in 5.x
- The 2.x plugin ``mapper-attachments`` was deprecated in 5.x in favour of [``ingest-attachment``](https://www.elastic.co/guide/en/elasticsearch/plugins/5.2/ingest-attachment.html)

We can merge the PR without having the tests pass on ES5, but the above issues needs to be fixed so that 5.x is installed correctly.

**Comment by slint** [comment]
Another issue (which unfortunately tests don't cover) is that the [Suggester API had some changes](https://www.elastic.co/guide/en/elasticsearch/reference/current/breaking_50_suggester.html) that will affect auto-completion fields in the deposit form. I think adding tests for features that depend on Elasticsearch (eg. OAI-PMH) is a necessary step to be able to proceed with upgrading to v5.x.

**Comment by rusrushal13** [comment]
As 2.x version doesn't support ingest attachment so is it necessary? @lnielsen @krzysztof

**Comment by krzysztof** [comment]
@rusrushal13 The plubin in 2.x it's called "mapper-attachments", in 5.x - "ingest-attachment". Depending on ES version, only either should be installed.

**Comment by rusrushal13** [comment]
@krzysztof So should I install mapper attachment plugin or not?

**Comment by krzysztof** [comment]
@rusrushal13 For 5.x no, for 2.x yes.

**Comment by krzysztof** [comment]
@rusrushal13 Sorry, I just realized that we cannot support both at the same time. In that case you don't need mapper attachment.

**Comment by rusrushal13** [comment]
@krzysztof in that case we need to remove the V2.x environment

**Comment by rusrushal13** [comment]
@krzysztof @lnielsen @slint As installing V5.x is giving problems too, so can you suggest any other alternative in order to get it done.

**Pull request closed without merging**